scripts/get_collection_statics.py

```diff
@@ -0,0 +1,242 @@
+import os
+from argparse import ArgumentParser
+
+import pandas as pd
+
+from config_search import SOLR_URL
+from ht_full_text_search.ht_full_text_query import HTFullTextQuery
+from ht_full_text_search.ht_full_text_searcher import HTFullTextSearcher
+
+statics_fields = {"language008_full": "Language dist all",
+                  "countryOfPubStr": "Place of Pub",
+                  "bothPublishDateRange": "Date Dist",
+                  "htsource": "Source Libs",
+                  "callnoletters": "LC Class_dist",
+                  "rights": "Public Domain Dist"}
+
+def get_category_name(category: str) -> str:
+
+    """Get the category name from the category number
+    :param category:
+    :return: Category name
+    """
+
+    if '-' in category:
+        category = category.split('-')[1]
+    return category
+
+def map_callnoletters(df, map_file_path):
+    """
+    Map the two-letter callnoletters fields with the descriptions in the map_call_number.properties file.
+    :param df: DataFrame containing the callnoletters field.
+    :param map_file_path: Path to the map_call_number.properties file.
+    :return: Updated DataFrame with mapped callnoletters.
+    """
+    # Read the map_call_number.properties file
+    with open(map_file_path, 'r', encoding='latin-1') as file:
+        lines = file.readlines()
+
+    # Parse the properties file into a dictionary
+    callno_map = {}
+    general_callno_map = {}
+    for pos in range(3, 445):
```

Comment:
This might be worth a comment here about why (3,445) (that is, those are the lines in the file with information about LC classes). In the future if we continue using this for statistics we might want to think about improving how that works, but I don't think it's worth the effort right now.

Reply:
I agree that this is very poorly written hard code for parsing the LC class files more easily and quickly. I will add a TODO line